### ISSUE TITLE
qemu monitor keep flickering when there is no association with external physical display

### ISSUE THREAD

**Issue by right-0903** [intel/Display-Virtualization-for-Windows-OS#14]
I have captured the screen, as you can see it below.

https://github.com/intel/Display-Virtualization-for-Windows-OS/assets/111179762/399b7dcc-76e7-4df0-b5f2-faaa09fd0813

**Comment by right-0903** [comment]
I use libvirt, but this still happens when I use qemu commandline

```bash
sudo qemu-system-x86_64 -m 8192 -enable-kvm -cpu host -smp cores=4,threads=2,sockets=1 \
-drive file=/mnt/win-vm/windows-vm/Win10-22h2.qcow2,format=qcow2,cache=none \
-device vfio-pci,host=0000:00:02.1 \
-drive file=/usr/share/edk2-ovmf/x64/OVMF_CODE.fd,if=pflash,format=raw,unit=0,readonly=on \
-drive file=~/.config/libvirt/qemu/nvram/win10-22h2_VARS.fd,if=pflash,format=raw,unit=1 \
-device virtio-vga,max_outputs=1,blob=true -display gtk,gl=on,full-screen=off \
-object memory-backend-memfd,id=mem1,size=8192M -machine memory-backend=mem1
```

the text in the center when flickering is `the output display is not active`, the not active one is 
display1: wired display connected to Microsoft basic display driver

**Comment by malayaku** [comment]
Hope you have taken below QEMU patches, as mentioned in Readme.txt

QEMU Patches: https://github.com/intel/iotg-yocto-ese-main/tree/master/recipes-devtools/qemu/qemu

**Comment by right-0903** [comment]
> Hope you have taken below QEMU patches, as mentioned in Readme.txt
> 
> QEMU Patches: https://github.com/intel/iotg-yocto-ese-main/tree/master/recipes-devtools/qemu/qemu

thanks, but which branch should I use, should I apply all of them? I use stable-8.1 now, I apply them by num order, then I get many reject, and many build errors

**Comment by right-0903** [comment]
ok, for the problem of inactive output, I fixed it, some patches have been merged into qemu master branch, we just need patch `0040-virtio-gpu-replace-the-surface-with-null-surface-onl.patch` to make inactive output disappear.

but which patches are necessary for other features like connectors? applying all aptches is conflicted.

**Comment by downor** [comment]
@right-0903 
All the patches are written against 8.0. We would need to rebase patches that are not upstreamed yet to get "connectors" work. 
I suggest to use 8.0 + patches below (https://github.com/intel/iotg-yocto-ese-main/tree/master/recipes-devtools/qemu/qemu)
if you want to try our setup as is.

           file://0001-ui-gtk-skip-drawing-guest-scanout-when-associated-VC.patch \
           file://0002-ui-gtk-set-the-ui-size-to-0-when-invisible.patch \
           file://0003-ui-gtk-reset-visible-flag-when-window-is-minimized.patch \
           file://0004-ui-gtk-Disable-the-scanout-when-a-detached-tab-is-cl.patch \
           file://0005-ui-gtk-Factor-out-tab-window-creation-into-a-separat.patch \
           file://0006-ui-gtk-Add-a-new-parameter-to-assign-connectors-moni.patch \
           file://0007-ui-gtk-unblock-gl-if-draw-submitted-already-or-fence.patch \
           file://0008-ui-gtk-skip-drawing-if-any-of-ctx-surface-image-don-.patch \
           file://0009-ui-gtk-skip-refresh-rendering-if-VC-is-invisible.patch \
           file://0010-ui-console-display-inactive-display-output-when-repl.patch \
           file://0011-virtio-gpu-replace-the-surface-with-null-surface-onl.patch \
           file://0012-hw-virtio-Adds-support-for-VIRTIO_GPU_FORMAT_R8G8B8A.patch \
           file://0013-virtio-gpu-VIRTIO_GPU_RESP_ERR_OUT_OF_MEMORY-if-udma.patch \
           file://0014-ui-gtk-configure-the-area-of-the-texture-correctly.patch \
           file://0015-ui-gtk-making-dmabuf-NULL-when-dmabuf-is-not-valid-a.patch \
           file://0016-virtio-gpu-udmabuf-create-udmabuf-for-blob-when-iov_.patch \
           file://0017-ui-gtk-negative-page_number-shouldn-t-be-processed.patch \
           file://0018-ui-gtk-skip-refresh-if-new-dmabuf-is-submitted.patch \
           file://0019-ui-gtk-egl-Make-sure-the-right-EGL-context-is-curren.patch \
           file://0020-ui-gtk-Handle-relative-mode-events-correctly-with-Wa.patch \
           file://0021-ui-gtk-Consider-the-scaling-factor-when-getting-the-.patch \
           file://0022-ui-gtk-gl-area-Don-t-forget-to-calculate-the-scaling.patch \
           file://0023-virtio-gpu-Provide-position-info-x-y-to-the-Guest.patch \
           file://0024-ui-gtk-Include-the-position-info-while-setting-the-u.patch \
           file://0025-virtio-gpu-call-dpy_gl_frame_counter-at-every-guest-.patch \
           file://0026-ui-gtk-adds-status-bar-for-expressing-ups-and-fps.patch \
           file://0027-usb-hid-added-new-type-for-touch-stylus.patch \
           file://0028-ui-egl-helpers-display-EGL-error-if-eglCreateImageKH.patch \
           file://0029-ui-gtk-Added-an-input-mode.patch \
           file://0030-qemu-remove-invisible-boundary-of-guest-mouse-cursor.patch \
           file://0031-virtio-gpu-io-mediator-for-blob-true.patch \
           file://0001-ui-gtk-fullscreen-control-on-all-QEMU-windows.patch \

**Comment by right-0903** [comment]
@downor I see, thanks. I will try it later.
